sdks/go/pkg/beam/core/runtime/harness/file_system_hooks.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package harness
```

Comment:
Don't put this into the harness package. It can live in the GCS package, registered to the the hook. The harness already triggers hooks on worker startup.

harnessopts does things in the harness package because they're options for the harness, but that separation doesn't need to exist for the gcs package.

Reply:
@lostluck could you review changes again?